### ISSUE TITLE
MySqlConnector.Logging.Microsoft.Extensions.Logging is not prefixing the category name like other loggers

### ISSUE THREAD

**Issue by KirillBorunov** [mysql-net/MySqlConnector#1080]
**Software versions**
MySqlConnector.Logging.Microsoft.Extensions.Logging version: 2.0.0

**Describe the bug**
The mentioned package is not prefixing the log categories.
So I see categories like "ConnectionPool" instead of "MySqlConnector.ConnectionPool".
Because of this, collisions are possible and I cannot filter all the MySqlConnector logs with single filter, need to know each possible category.

Here is the missing prefix:

https://github.com/mysql-net/MySqlConnector/blob/44d5c266de3009b8d0afe668eeff4e3e31244b3a/src/MySqlConnector.Logging.Microsoft.Extensions.Logging/MicrosoftExtensionsLoggingLoggerProvider.cs#L11

The other logging packages DO prefix:

https://github.com/mysql-net/MySqlConnector/blob/44d5c266de3009b8d0afe668eeff4e3e31244b3a/src/MySqlConnector.Logging.log4net/Log4netLoggerProvider.cs#L11

https://github.com/mysql-net/MySqlConnector/blob/44d5c266de3009b8d0afe668eeff4e3e31244b3a/src/MySqlConnector.Logging.Serilog/SerilogLoggerProvider.cs#L16

https://github.com/mysql-net/MySqlConnector/blob/44d5c266de3009b8d0afe668eeff4e3e31244b3a/src/MySqlConnector.Logging.NLog/NLogLoggerProvider.cs#L9

**Exception**
Not applicable.

**Code sample**

```csharp
LoggerFactory = Microsoft.Extensions.Logging.LoggerFactory.Create(cfg => 
    cfg.AddConsole()
    .SetMinimumLevel(LogLevel.Debug)
    .AddFilter("ConnectionPool", LogLevel.Information) //How can I filter all MySqlConnector logs? How I can know all posible categories?
);

MySqlConnectorLogManager.Provider = new MicrosoftExtensionsLoggingLoggerProvider(LoggerFactory);
```

**Expected behavior**
I expect a prefix "MySqlConnector." just like on the other loggers.

**Additional context**
None

**Comment by bgrainger** [comment]
I think you're right, but it's surprising no one's pointed this out before. (And now it would be a breaking behavioural change... so it would have been nice to know a week ago, before shipping 2.0. 😀)

For now, it's fairly simple to work around with your own code:

```csharp
sealed class MySqlConnectorPrefixLoggerFactory : ILoggerFactory
{
	readonly ILoggerFactory _factory;
	
	public MySqlConnectorPrefixLoggerFactory(ILoggerFactory factory) => _factory = factory;
	
	public void AddProvider(ILoggerProvider provider) => _factory.AddProvider(provider);

	public ILogger CreateLogger(string categoryName) => _factory.CreateLogger("MySqlConnector." + categoryName);

	public void Dispose() => _factory.Dispose();
}

// ... other code

MySqlConnectorLogManager.Provider = new MicrosoftExtensionsLoggingLoggerProvider(new MySqlConnectorPrefixLoggerFactory(loggerFactory));
```

(Does such a category-name-prefixing type already exist somewhere? It seems like a useful type but I couldn't find one after a brief search.)

I'll have to think about how best to implement this change (to avoid upgrade surprises from 1.0.0 -> 2.0.0 -> fixed version).

**Comment by KirillBorunov** [comment]
> For now, it's fairly simple to work around with your own code:

That's exactly what I did. It works OK.

> I'll have to think about how best to implement this change (to avoid upgrade surprises from 1.0.0 -> 2.0.0 -> fixed version).

What if you allow to at least specify a OPTIONAL prefix on the constructor?
Maybe on all the logger packages. Maybe someone wants to use a category "DB" to log all database related messages the MySqlConnector ones + custom ones.
That way you will not fix it directly but add a feature that will allow users to do what they want with categories/prefixes.

```
new MicrosoftExtensionsLoggingLoggerProvider(LoggerFactory, "DB"); //constructor should ensure there is a dot
```

**Comment by bgrainger** [comment]
I was thinking an opt-out, e.g., `MicrosoftExtensionsLoggingLoggerProvider(ILoggerFactory loggerFactory, bool omitMySqlConnectorPrefix = false);`

This would give better logging by default, but allow users who were relying on the old category names to opt out.

**Comment by KirillBorunov** [comment]
Your way seems easier to implement, and solves the original problem, I think its OK.

**Comment by bgrainger** [comment]
Fixed in [2.1.0](https://github.com/mysql-net/MySqlConnector/releases/tag/2.1.0).
